babel/localedata.py

```diff
@@ -24,15 +25,29 @@
 _dirname = os.path.join(os.path.dirname(__file__), 'locale-data')
 
 
+def normalize_locale(name):
+    """Normalize a locale ID by stripping spaces and apply proper casing.
+
+    Returns the normalized locale ID string or `None` if the ID is not
+    recognized.
+    """
+    name = name.strip().lower()
+    for locale_id in chain.from_iterable([_cache, locale_identifiers()]):
+        if name == locale_id.lower():
+            return locale_id
+
+
 def exists(name):
```

Comment:
Wow, greedy method.
I'm really looking forward to https://github.com/python-babel/babel/pull/305 !

(just an observation, please ignore :)